### ISSUE TITLE
Implement 404 Error state for users signed into Cal-ITP Benefits Administrator 

### ISSUE THREAD

**Issue by indexing** [cal-itp/benefits#2387]
The Cal-ITP Benefits application currently has a [404 error view](https://www.figma.com/design/SeSd3LaLd6WkbEYhmtKpO3/Benefits-(Full-Application)?m=auto&node-id=9234-104877&t=CYTmdJcjM77GlOpn-1). 

The current 404 error view has a couple problems now that we have introduced the admin interface and a new set of users.
- Its styling aligns with the rider-facing application. 
- The `return home` button goes to the rider-facing home view. 

In a nutshell, the design is intended for transit riders only; it doesn't account for admin users.

We need a 404 error view for admin users that have authenticated with the application.

**Comment by thekaveman** [comment]
Related #3118
